vowpalwabbit/config/tests/options_cli_test.cc

```diff
@@ -2,27 +2,22 @@
 // individual contributors. All rights reserved. Released under a BSD (revised)
 // license as described in the file LICENSE.
 
-#include <boost/test/unit_test.hpp>
-#include <boost/test/test_tools.hpp>
-#include <boost/mpl/vector.hpp>
+#include "vw/config/options_cli.h"
 
-#include "config/options_cli.h"
-#include "test_common.h"
+#include "vw/config/cli_options_serializer.h"
 
-#include "memory.h"
-#include "config/cli_options_serializer.h"
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
 
 #include <memory>
 #include <vector>
 
 using namespace VW::config;
 
-using option_types = boost::mpl::vector<options_cli>;
-
-BOOST_AUTO_TEST_CASE_TEMPLATE(typed_options_parsing, T, option_types)
+TEST(options_cli_test, typed_options_parsing)
 {
   std::vector<std::string> args = {"--str_opt", "test_str", "-i", "5", "--bool_opt", "--float_opt", "4.3"};
-  auto options = VW::make_unique<T>(args);
+  auto options = std::unique_ptr<options_cli>(new options_cli(args));
```

Comment:
Why this change?

Reply:
The boost implementation is gone and so the test file no longer needs to be templated over two types. Its a lot simpler to just use 1 type so I moved back to that